gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2442,4 +2455,23 @@ public Map<String, Long> getStatistics() {
     return statistics.entrySet().stream()
         .collect(toImmutableMap(Map.Entry::getKey, e -> e.getValue().get()));
   }
+
+  private String getUgiUserName() {
+    if ((!storageOptions.getHttpRequestHeaders().containsKey(CUSTOM_AUDIT_USER_HEADER))
+        && storageOptions.isGcsAuditLogEnabled()) {
+      return getCurrentLoggedInUser();
+    }
+    return null;
+  }
+
+  private static String getCurrentLoggedInUser() {
+    try {
+      UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+      return ugi.getShortUserName();
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log(
+          "Ignoring IOException; failure to fetch current user info");
+      return null;
```

Comment:
Let us re throw the exception if this is not expected to fail. That way we will not hide the bugs.

Reply:
We do not want workflows to fail which were working fine before in case a functionality that we have added fails.

Reply:
Not all workflows will be affected correct. Only if they had the audit flag enabled, only then it will hit this code path, correct?